uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/installer/PlayStore.java

```diff
@@ -150,18 +150,31 @@ private void installOrUpdateAppFromMarketPageInternal(String playStoreAction) th
                 new UiSelector().className(Button.class).text(playStoreAction).enabled(true)
         );
 
-        uiObjBtn.waitForExists(FIND_UI_ELEMENT_TIMEOUT);
+        // Sometimes, firebase devices take a very long time to download an app (outlook for example),
+        // a subsequent retry on the same test will find the app already downloading, and will not find the install/update button
+        // instead of just checking for install/update button, we can check for cancel button (Without pressing it)
```

Comment:
I'm not entirely sure I understand what's happening here. Are we speeding up the test?

Reply:
Not really speeding it up, just improving the chances for next retry to pass. What happens is firebase sometimes gives us a device with questionable download speed, so downloading the application takes longer than 5 minutes. By the time the retry starts, it's still downloading, so we check for the cancel button incase it's still downloading by the time we reach the play store page again. We could argue against doing this since we want a "clean" slate at the start of each test, but I also think it might be a waste to run these playstore test cases on firebase since they are quite costly budget wise. I have suggestion I got from OneAuth team about how we could alleviate this. Will include it in the upcoming design doc.

Reply:
I'm going to get rid of this, this is a bad solution regardless of what we decide.